pyiron_atomistics/atomistics/structure/factories/compound.py

```diff
@@ -0,0 +1,189 @@
+
+_ase = AseFactory()
+
+
+def _bcc_lattice_constant_from_nn_distance(element):
+    """
+    Build a BCC lattice constant by making the BCC have the same nearest neighbour distance as the regular cell.
+
+    Works because the NN distance doesn't even care what crystal structure the regular unit cell is.
+    """
+    return AseFactory().bulk(name=element).get_neighbors(num_neighbors=1).distances[0, 0] * (2 / np.sqrt(3))
+
+
+class CompoundFactory:
+    """A collection of routines for constructing Laves phases and other more complex structures."""
+
+    @staticmethod
+    def B2(element_a, element_b, a=None):
+        """
+        Builds a cubic $AB$ B2 structure of interpenetrating simple cubic lattices.
+
+        Args:
+            element_a (str): The chemical symbol for the A element.
+            element_b (str): The chemical symbol for the B element.
+            a (float): The lattice constant. (Default is None, which uses the default for element A.)
+
+        Returns:
+            (Atoms): The B2 unit cell.
+        """
+        a = _bcc_lattice_constant_from_nn_distance(element_a) if a is None else a
+
+        return _ase.crystal((element_a, element_b), [(0, 0, 0), (1/2, 1/2, 1/2)], spacegroup=221, cell=(a, a, a))
+
+    @staticmethod
+    def C14(element_a, element_b, a=None, c_over_a=1.626, x1=0.5, z1=0.5):
+        """
+        Builds a hexagonal $A B_2$ C14 Laves phase cell.
+
+        .. attention:: Change in Stochiometry possible!
+
+            If any of the fractional coordinates fall onto their high symmetry values the atoms may be placed on another
+            Wyckoff position, with less sites and therefor the cell composition may change.
+
+            For `x1` avoid 0, 1/3, 2/3, for `z1` avoid 1/4, 3/4.
+
+        Args:
+            element_a (str, ase.Atom): specificies A
+            element_b (str, ase.Atom): specificies B
+            a (float): length of a & b cell vectors
+            c_over_a (float): c/a ratio
+            x1 (float): fractional x coordinate of B atoms on Wyckoff 6h
+            z1 (float): fractional z coordinate of A atoms on Wyckoff 4f
+        """
+        a = 2 * _bcc_lattice_constant_from_nn_distance(element_a) if a is None else a
+        c = c_over_a * a
+
+        # https://www.atomic-scale-physics.de/lattice/struk/c14.html
+        s = _ase.crystal(( element_a, element_b, element_b ),
+                            [ # wyckoff 4f -- A
+                              ( 1/3,   2/3,  z1 ),
+                              # wyckoff 2a -- B-I
+                              (   0,     0,   0 ),
+                              # wyckoff 6h -- B-II
+                              ( 1*x1, 2*x1, 1/4 ),
+                            ],
+                            spacegroup=194, cell=(a, a, c, 90, 90, 120))
+        if len(s) != 2 + 6 + 4:
+            raise ValueError("Given internal coordinates reduced symmetry, check the docstring for degenerate values!")
+        return s
```

Comment:
Hello, 
I didn't obtained a correct c14 structure with this, to many Mg atoms were in the triple layer and the Ca atoms were not good located as well. By using these values (1/3, 2/3, 0.5629),(0,0,0), (0.1697,0.3394,1/4), it works well :)
I found the correct value here: https://materials.springer.com/isp/crystallographic/docs/sd_1822295

Reply:
Do you get your intended structure if you specify `x1=0.1697` and `z1=0.5629`? If yes, we may argue about the defaults, here, but the general function is working.

Reply:
I use directly these lines of the code:
`c14=pr.create.structure.ase.crystal(('Ca','Mg','Mg'), 
                             ((1/3, 2/3, 0.5629),(0,0,0), (0.1697,0.3394,1/4)),
                            spacegroup=194, cell=(6.146, 6.146, 9.979, 90, 90, 120)).repeat([3,3,3])`
So I guess the general functions is working, but x1 and z1 have to change to reproduce the c14 correctly :)

Reply:
I updated the defaults, also the ones for C36.

Reply:
The C36 structure seems good now for me with the new values, as well as the C15.